### ISSUE TITLE
no-irregular-whitespace add additional chars

### ISSUE THREAD

**Issue by mattrabe** [eslint/eslint#11800]
**What rule do you want to change?**

no-irregular-whitespace

**Does this change cause the rule to produce more or fewer warnings?**

More

**How will the change be implemented? (New option, new default behavior, etc.)?**

New default behavior - add to list of irregulars

**Please provide some example code that this change will affect:**

The following contains \u0003:

```js
Any content that contains a gremlin character.
```

**What does the rule currently do for this code?**

Nothing

**What will the rule do after it's changed?**

Warn on additional list of irregular chars

**Are you willing to submit a pull request to implement this change?**

Absolutely

We are experiencing "gremlin" characters getting introduced into our codebase. We believe they are originating from copying content out of design software such as Photoshop, though we are not 100% certain of the origin. This rule is not clearing some of the irregular chars. In particular, we are seeing `\u0003` creep in, and it is not cleared by this rule.

We also utilize the VSCode plugin [Gremlins tracker for Visual Studio Code](https://marketplace.visualstudio.com/items?itemName=nhoizey.gremlins) which does a good job of notifying us of these "gremlin" chars. Some, but not all, of the chars that this plugin recognizes as "gremlin" are currently fixed by this rule... Looking at the source of the VSCode extension, the following additional irregular chars could be added to this rule:

```
\u0003
\u200c
\u200e
\u201c
\u201d
\u202c
\u202d
\u202e
\ufffc
```

I am no expert on unicode, but I know that we are seeing at least \u0003 getting added to our codebase on a somewhat regular basis.

**Comment by mattrabe** [comment]
For what it's worth I have create an eslint plugin package which includes this rule as `no-gremlin-chars`: https://www.npmjs.com/package/eslint-plugin-fpcs

My team will use this for the time being, until/unless `no-irregular-whitespace` is updated to include these additional chars. I would be happy to open a PR for that

**Comment by mattrabe** [comment]
pull request #11907
